File: backend/components/metiers/administration_publique/fixtures/helpers/validators/test_validators.js
// test_validators.js – Tests unitaires et edge cases validators JS
const validators = require('./validators');

test('isValid3DModel returns true for valid model', () => {
  expect(validators.isValid3DModel({ id: 'x', vertices: [] })).toBe(true);
  expect(validators.isValid3DModel({ id: 1, vertices: null })).toBe(false);
});

test('isValidUser returns true for valid user', () => {
  expect(validators.isValidUser({ id: 'u', role: 'admin' })).toBe(true);
  expect(validators.isValidUser({ id: 1, role: null })).toBe(false);
});

test('isFixtureAccessible returns true for accessible fixture', () => {
  expect(validators.isFixtureAccessible({ description: 'ok' })).toBe(true);
  expect(validators.isFixtureAccessible({ description: 123 })).toBe(false);
});

File: backend/components/metiers/threed/fixtures/helpers/validators/test_validators.py
import importlib

def test_is_valid_3d_model():
    validators = importlib.import_module('backend.components.metiers.threed.fixtures.helpers.validators')
    assert validators.is_valid_3d_model({'id': 'x', 'vertices': []})
    assert not validators.is_valid_3d_model({'id': 1, 'vertices': None})

def test_is_valid_user():
    validators = importlib.import_module('backend.components.metiers.threed.fixtures.helpers.validators')
    assert validators.is_valid_user({'id': 'u', 'role': 'admin'})
    assert not validators.is_valid_user({'id': 1, 'role': None})

def test_is_fixture_accessible():
    validators = importlib.import_module('backend.components.metiers.threed.fixtures.helpers.validators')
    assert validators.is_fixture_accessible({'description': 'ok'})
    assert not validators.is_fixture_accessible({'description': 123})
